src/Main.hs

```diff
@@ -9,27 +9,28 @@ import HigherLower.Game
 main :: IO ()
 main = do
   g <- createGame
---  runState (mfix play) $ asStateTIO g
-  return ()
+  evalStateT playGame g
```

Comment:
runStateT would give you the new state as well as the result of the playGame computation, which is (), in a nice tuple.
evalStateT and execStateT lets you pick the fst or snd of the tuple (depending which one you're interested in).

In our case, we really care about neither, so I just arbitrarily choose eval which gives the result but then discard the result by not binding it to any name.